codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpProtocolGenerator.kt

```diff
@@ -579,13 +576,23 @@ private class ServerHttpProtocolImplGenerator(
                 *codegenScope,
                 "parser" to parser,
             )
+        } else if (streamingMember != null) {
+            rustTemplate(
+                """
+                let body = request.take_body().ok_or(#{SmithyHttpServer}::rejection::BodyAlreadyExtracted)?;
+                input = input.${streamingMember.setterName()}(Some(body.into()));
+                """.trimIndent(),
+                *codegenScope
+            )
         }
-        for (binding in bindings) {
-            val member = binding.member
-            val parsedValue = serverRenderBindingParser(binding, operationShape, httpBindingGenerator, structuredDataParser)
-            if (parsedValue != null) {
-                withBlock("input = input.${member.setterName()}(", ");") {
-                    parsedValue(this)
+        if (streamingMember == null) {
```

Comment:
I wanted to highlight this condition I added - I had to check it so that with streaming there will be no bindings parsers generated. The weird thing is just the flow of this function - the conditions seems a bit detached from one another but I wasn't sure if there's a way to simplify:
```kotlin
if (parser != null) { ... } 
else if (streamingMember != null) { ... }
if (streamingMember == null) { ... }
```

Reply:
@guymguym Thanks for calling this out. We still need to generate HTTP bindings for blob streaming (e.g. HTTP header bindings). This also led me to discover there's an error in Smithy that is now being fixed over at https://github.com/awslabs/smithy/issues/1075

I have pushed a new commit to this branch, [Move blob streaming handling to httpPayload binding generation](https://github.com/awslabs/smithy-rs/pull/1023/commits/d86443ee023e3685622df9f9c96971b7fed8eef8), that refactors this part and generates the stream handling when we handle `@httpPayload`.

Let me know if you agree with the fix or have any questions.

Reply:
Ah now I see. I think my problem was probably that the previous headers code that I used was not very accurate, but since your PR to implement server response headers that might have fixed this issue. I will test a bit later and report back if I still see any issues in the S3 server codegen, but it looks good to me. Thanks!